### PR TITLE
Put request

### PR DESCRIPTION
**Pull request by kristjanreba**
Add put request function as part of redfish client.

### CODE REVIEW

**Comment by kristjanreba** [comment]
Resolves: #27

File: tests/test_resource.py [hunk 1]
```diff
@@ -181,6 +181,21 @@ def test_patch_an_invalid_resource(self):
             Resource(connector, data={}).patch(payload={"Misc": "MyValue"})
 
 
+class TestPut:
+    def test_put_a_resource(self):
+        connector = mock.Mock(spec=Connector)
+        Resource(connector, data={
+            "@odata.id": "/redfish/v1/Systems/1"
+        }).put(payload={"Misc": "MyValue"})
```
Comment (tadeboro) [inline tests/test_resource.py:189]: We are missing one more test that will test behavior when the `path` parameter is set.

**Review by tadeboro — CHANGES_REQUESTED** [review]
Thank you, @kristjanreba, for your contribution. The code looks great, but we are missing one test. Apart from that, things are ready to go.

File: tests/test_resource.py (hunk 1 above)
Comment (kristjanreba) [reply tests/test_resource.py:189]: Thank you for spotting the missing test. It is now included.

File: redfish_client/resource.py [hunk 2]
```diff
@@ -210,6 +215,19 @@ def patch(self, payload):
         if not path:
             raise MissingOidException("The resource cannot be PATCHed.")
         return self._connector.patch(path, payload=payload)
+    
+    def put(self, path=None, payload=None):
+        """
+        Perform a PUT at the resource with the given payload.
```
Comment (tadeboro) [inline redfish_client/resource.py:221]:
```suggestion
        Perform a PUT at the resource or selected path with the given payload.
```

File: redfish_client/resource.py [hunk 3]
```diff
@@ -210,6 +215,19 @@ def patch(self, payload):
...
+
+        Args:
+          payload: The contents of the PUT payload.
```
Comment (tadeboro) [inline redfish_client/resource.py:224]: We are missing the path argument docs.

**Review by tadeboro — CHANGES_REQUESTED** [review]
Almost there ;)

**Review by tadeboro — APPROVED** [review]
Thank you, @kristjanreba, for persevering! ;)

**Pull request merged**